test/ProductConstructionService.SubscriptionTriggerer.Tests/SubscriptionTriggererTests.cs

```diff
@@ -0,0 +1,335 @@
+            AzureDevOpsBuildNumber = "old.build.number",
+            Commit = "oldSha",
+            DateProduced = DateTimeOffset.UtcNow.AddDays(-2)
+        };
+        var location = "https://source.feed/index.json";
+        Build build = new()
+        {
+            AzureDevOpsBranch = "source.branch",
+            AzureDevOpsRepository = "source.repo",
+            AzureDevOpsBuildNumber = "build.number",
+            Commit = "sha",
+            DateProduced = DateTimeOffset.UtcNow,
+            Assets =
+            [
+                new Asset
+                {
+                    Name = "source.asset",
+                    Version = "1.0.1",
+                    NonShipping = false,
+                    Locations =
+                    [
+                        new AssetLocation
+                        {
+                            Location = location,
+                            Type = LocationType.NugetFeed
+                        }
+                    ]
+                }
+            ]
+        };
+        BuildChannel buildChannel = new()
+        {
+            Build = build,
+            Channel = channel
+        };
+        Subscription subscription = new()
+        {
+            Channel = channel,
+            SourceRepository = "source.repo",
+            TargetRepository = "target.repo",
+            TargetBranch = "target.branch",
+            Enabled = true,
+            PolicyObject = new SubscriptionPolicy
+            {
+                MergePolicies = null,
+                UpdateFrequency = UpdateFrequency.EveryWeek
+            },
+            LastAppliedBuild = oldBuild
+        };
+        Repository repoInstallation = new()
+        {
+            RepositoryName = "target.repo",
+            InstallationId = 1
+        };
+        await _context!.Repositories.AddAsync(repoInstallation);
+        await _context!.Subscriptions.AddAsync(subscription);
+        await _context!.BuildChannels.AddAsync(buildChannel);
+        await _context!.SaveChangesAsync();
+
+        var triggerer = ActivatorUtilities.CreateInstance<SubscriptionTriggerer>(_scope.ServiceProvider);
+        List<UpdateSubscriptionWorkItem> list = await triggerer.GetSubscriptionsToTrigger(UpdateFrequency.EveryDay);
+
+        list.Count.Should().Be(0);
+    }
+
+    [Test]
+    public async Task ShouldNotTriggerUpToDateSubscription()
+    {
+        var channel = new Channel
+        {
+            Name = "channel",
+            Classification = "class"
+        };
+        var build = new Build
+        {
+            AzureDevOpsBranch = "source.branch",
+            AzureDevOpsRepository = "source.repo",
+            AzureDevOpsBuildNumber = "build.number",
+            Commit = "sha",
+            DateProduced = DateTimeOffset.UtcNow
+        };
+        var buildChannel = new BuildChannel
+        {
+            Build = build,
+            Channel = channel
+        };
+        var subscription = new Subscription
```

Comment:
Could at least some of this test data be instantiated once per test class? Having it in defined in every single test makes them a bit bulky and harder to see the params we want to focus on

Reply:
There's a part that's shared between the tests, but It might not be that clean to read, couse you have to look in two places. It is bulky, but it's all in one place. I can extract it in methods tho

Reply:
I also deleted some stuff that wasn't needed